src/security-council-mgmt/SecurityCouncilManager.sol

```diff
@@ -237,26 +252,36 @@ contract SecurityCouncilManager is
         external
         onlyRole(DEFAULT_ADMIN_ROLE)
     {
-        // CHRIS: TODO: do we also want to check for duplicates when adding here?
         _addSecurityCouncil(_securityCouncilData);
     }
 
     /// @inheritdoc ISecurityCouncilManager
-    function removeSecurityCouncil(uint256 _index) external onlyRole(DEFAULT_ADMIN_ROLE) {
-        // CHRIS: TODO: I wonder if this should be based on address, or some other identifying behaviour?
-        SecurityCouncilData storage securityCouncilToRemove = securityCouncils[_index];
-        SecurityCouncilData storage lastSecurityCouncil =
-            securityCouncils[securityCouncils.length - 1];
-
-        securityCouncils[_index] = lastSecurityCouncil;
-        securityCouncils.pop();
-        emit SecurityCouncilRemoved(
-            securityCouncilToRemove.securityCouncil,
-            // securityCouncilToRemove.upgradeExecutor,
-            securityCouncilToRemove.updateAction,
-            // securityCouncilToRemove.inbox,
-            securityCouncils.length
-        );
+    function removeSecurityCouncil(SecurityCouncilData memory _securityCouncilData)
+        external
+        onlyRole(DEFAULT_ADMIN_ROLE)
+        returns (bool)
+    {
+        for (uint256 i = 0; i < securityCouncils.length; i++) {
+            SecurityCouncilData storage securityCouncilData = securityCouncils[i];
+            if (
+                securityCouncilData.securityCouncil == _securityCouncilData.securityCouncil
+                    && securityCouncilData.chainId == _securityCouncilData.chainId
+                    && securityCouncilData.updateAction == _securityCouncilData.updateAction
+            ) {
+                SecurityCouncilData storage lastSecurityCouncil =
+                    securityCouncils[securityCouncils.length - 1];
+
+                securityCouncils[i] = lastSecurityCouncil;
+                securityCouncils.pop();
+                emit SecurityCouncilRemoved(
+                    securityCouncilData.securityCouncil,
+                    securityCouncilData.updateAction,
+                    securityCouncils.length
+                );
+                return true;
```

Comment:
We dont need to return bool here, since we revert instead of false

Reply:
the return is there to short circuit so that revert outside of the for loop never executes